commands/registry.go

```diff
@@ -0,0 +1,244 @@
+}
+
+const registryHostname = "registry.digitalocean.com"
+
+// Registry creates the registry command
+func Registry() *Command {
+	cmd := &Command{
+		Command: &cobra.Command{
+			Use:     "registry",
+			Aliases: []string{"reg", "r"},
+			Short:   "[beta] registry commands",
+			Long:    "[beta] registry is used to access container registry commands",
+			Hidden:  true,
+		},
+	}
+
+	CmdBuilder(cmd, RunRegistryCreate, "create <registry-name>", "create container registry", Writer)
+
+	CmdBuilder(cmd, RunRegistryGet, "get", "get the container registry", Writer, aliasOpt("g"), displayerType(&displayers.Registry{}))
+
+	cmdRunRegistryDelete := CmdBuilder(cmd, RunRegistryDelete, "delete", "delete the container registry", Writer, aliasOpt("del"))
+	AddBoolFlag(cmdRunRegistryDelete, doctl.ArgForce, doctl.ArgShortForce, false, "Force registry delete")
+
+	CmdBuilder(cmd, RunRegistryLogin, "login", "log in Docker to the container registry", Writer)
+	CmdBuilder(cmd, RunRegistryLogout, "logout", "log out Docker from the container registry", Writer)
+
+	cmdRunKubernetesManifest := CmdBuilder(cmd, RunKubernetesManifest, "kubernetes-manifest", "create a Kubernetes secret manifest to allow read/pull access to the registry", Writer, aliasOpt("k8s"))
+	AddStringFlag(cmdRunKubernetesManifest, doctl.ArgObjectName, "", "", "the secret name to create. defaults to the registry name prefixed with \"registry-\"")
+	AddStringFlag(cmdRunKubernetesManifest, doctl.ArgObjectNamespace, "", "default", "the namespace to hold the secret")
+
+	return cmd
+}
+
+// Registry
+
+// RunRegistryCreate creates a registry
+func RunRegistryCreate(c *CmdConfig) error {
+	if len(c.Args) != 1 {
+		return doctl.NewMissingArgsErr(c.NS)
+	}
+
+	name := c.Args[0]
+	rs := c.Registry()
+
+	rcr := &godo.RegistryCreateRequest{Name: name}
+	r, err := rs.Create(rcr)
+	if err != nil {
+		return err
+	}
+
+	return displayRegistries(c, *r)
+}
+
+// RunRegistryGet returns the registry
+func RunRegistryGet(c *CmdConfig) error {
+	reg, err := c.Registry().Get()
+	if err != nil {
+		return err
+	}
+
+	return displayRegistries(c, *reg)
+}
+
+// RunRegistryDelete delete the registry
+func RunRegistryDelete(c *CmdConfig) error {
+	force, err := c.Doit.GetBool(c.NS, doctl.ArgForce)
+	if err != nil {
+		return err
+	}
+
+	if !force && AskForConfirm("delete registry") != nil {
+		return fmt.Errorf("operation aborted")
+	}
+
+	return c.Registry().Delete()
+}
+
+// DockerConfigProvider allows a user to read from a remote and local Kubeconfig, and write to a
+// local Kubeconfig.
+type DockerConfigProvider interface {
+	ConfigPath() string
+}
+
+// store execCommand in a variable. Lets us override it while testing
+var execCommand = exec.Command
+
+// RunRegistryLogin logs in Docker to the registry
+func RunRegistryLogin(c *CmdConfig) error {
+	creds, err := c.Registry().DockerCredentials(&godo.RegistryDockerCredentialsRequest{
+		ReadWrite: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	dc := &dockerConfig{}
```

Comment:
nit: can this please be `var dockerConfig`. The declare and assign isn't actually needed. You would need to put the `&` below on your `dc` in json.Unmarshal.

Reply:
👍🏼